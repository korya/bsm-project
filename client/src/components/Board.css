/* Based on https://codepen.io/artemisart/pen/NNErXO */

div.sudoku-board-empty {
  width: 386px;
  height: 386px;
  border: 1px solid black;
  margin: 20px auto;
  background-color: #d6d6d6;
}

table.sudoku-board {
  margin: 20px auto;
  border-collapse: collapse;
}

table.sudoku-board tr:first-child td {
  border-top-color: black;
}

table.sudoku-board tr:nth-child(3n) td {
  border-bottom-color: black;
}

table.sudoku-board td {
  border: 1px solid lightgrey;
  height: 40px;
  width: 40px;
}

table.sudoku-board td:first-child {
  border-left-color: black;
}

table.sudoku-board td:nth-child(3n) {
  border-right-color: black;
}

table.sudoku-board td button {
  padding: 0;
  text-align: center;
  border: 0;
  height: 40px;
  width: 40px;
  text-align: center;
  font-size: 16px;
  cursor: pointer;
}

table.sudoku-board td button.selected {
  background-color: rgba(114, 221, 253, 0.63);
}

table.sudoku-board td button:focus {
  outline: 0;
}
