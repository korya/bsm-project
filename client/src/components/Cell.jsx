import React from 'react';

export default function Cell({ number, selected, onClick }) {
  const className = selected ? 'selected' : '';

  return (
    <td>
      <button
        className={className}
        onClick={onClick}
      >
        {number}
      </button>
    </td>
  );
}
