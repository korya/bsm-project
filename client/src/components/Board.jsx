import React from 'react';
import Cell from './Cell';
import './Board.css';

export default function Board({ board, selected, onCellClick }) {
  return (
    <table className="sudoku-board">
      <tbody>
        {
          seq(9).map(i => (
            <tr key={i}>
              {
                seq(9).map(j => {
                  const idx = i * 9 + j;
                  return (
                    <Cell
                      key={idx}
                      number={board[idx]}
                      selected={selected[idx]}
                      onClick={() => onCellClick(idx)}
                    />
                  );
                })
              }
            </tr>
          ))
        }
      </tbody>
    </table>
  );
}

const seq = length => Array.from({ length }).map((n, idx) => idx);
