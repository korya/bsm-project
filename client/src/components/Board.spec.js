import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Board from './Board';
import Cell from './Cell';

describe('renders correctly', () => {
  test('no selected cells', () => {
    const b = getTestBoard();
    const s = b.map(() => false);
    expect(
      renderer.create(
        <Board board={b} selected={s} />
      ).toJSON(),
    ).toMatchSnapshot();
  });

  test('with selected cells', () => {
    const b = getTestBoard();
    const s = b.map(() => false);
    s[1] = true;
    s[56] = true;
    s[80] = true;
    expect(
      renderer.create(
        <Board board={b} selected={s} />
      ).toJSON(),
    ).toMatchSnapshot();
  });
});

test('renders Cells', () => {
  const b = getTestBoard();
  const s = b.map(() => false);
  s[0] = true;
  s[2] = true;
  s[56] = true;
  s[80] = true;

  const wrapper = shallow(<Board board={b} selected={s} />);
  expect(
    wrapper.find(Cell).map(node => node.prop('number')),
  ).toEqual(b);
  expect(
    wrapper.find(Cell).map(node => node.prop('selected')),
  ).toEqual(s);
});

test('calls onCellClick prop handler', () => {
  const clickHandler = jest.fn();
  const b = getTestBoard();
  const s = b.map(() => false);
  const wrapper = shallow(
    <Board board={b} selected={s} onCellClick={clickHandler} />
  );
  wrapper.find(Cell).find({ number: 112 }).simulate('click');
  expect(clickHandler).toHaveBeenCalledTimes(1);
  expect(clickHandler).toHaveBeenCalledWith(12);
});

// A board is not valid, it contains 100+index in every cell. But it is
// very handy for testing.
const getTestBoard = () => Array.from({ length: 81 }).map((n, idx) => 100+idx);
