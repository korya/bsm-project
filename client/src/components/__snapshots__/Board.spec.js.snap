// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly no selected cells 1`] = `
<table
  className="sudoku-board"
>
  <tbody>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          100
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          101
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          102
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          103
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          104
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          105
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          106
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          107
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          108
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          109
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          110
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          111
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          112
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          113
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          114
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          115
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          116
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          117
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          118
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          119
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          120
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          121
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          122
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          123
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          124
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          125
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          126
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          127
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          128
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          129
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          130
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          131
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          132
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          133
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          134
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          135
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          136
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          137
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          138
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          139
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          140
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          141
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          142
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          143
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          144
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          145
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          146
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          147
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          148
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          149
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          150
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          151
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          152
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          153
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          154
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          155
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          156
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          157
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          158
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          159
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          160
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          161
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          162
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          163
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          164
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          165
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          166
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          167
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          168
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          169
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          170
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          171
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          172
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          173
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          174
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          175
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          176
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          177
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          178
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          179
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          180
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders correctly with selected cells 1`] = `
<table
  className="sudoku-board"
>
  <tbody>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          100
        </button>
      </td>
      <td>
        <button
          className="selected"
          onClick={[Function]}
        >
          101
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          102
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          103
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          104
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          105
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          106
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          107
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          108
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          109
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          110
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          111
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          112
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          113
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          114
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          115
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          116
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          117
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          118
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          119
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          120
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          121
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          122
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          123
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          124
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          125
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          126
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          127
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          128
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          129
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          130
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          131
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          132
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          133
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          134
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          135
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          136
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          137
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          138
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          139
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          140
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          141
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          142
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          143
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          144
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          145
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          146
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          147
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          148
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          149
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          150
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          151
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          152
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          153
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          154
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          155
        </button>
      </td>
      <td>
        <button
          className="selected"
          onClick={[Function]}
        >
          156
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          157
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          158
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          159
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          160
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          161
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          162
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          163
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          164
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          165
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          166
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          167
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          168
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          169
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          170
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          171
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          172
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          173
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          174
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          175
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          176
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          177
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          178
        </button>
      </td>
      <td>
        <button
          className=""
          onClick={[Function]}
        >
          179
        </button>
      </td>
      <td>
        <button
          className="selected"
          onClick={[Function]}
        >
          180
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
